packages/orbit-tailwind-preset/src/presets/components/index.ts

```diff
@@ -66,191 +70,194 @@ const getBackgroundColors = () => {
 
 const cfg = (options?: Options): Config => {
   const { disablePreflight = true } = options || {};
+  // make palette colors (foundation colors) defined as css vars and make components tokens inherit it
+  const theme = getTokens(cssVarsFoundation);
```

Comment:
wdym by refactoring soon after this? there is no way how to resolve it without `getTokens` function that is documented in tokens package and basically the only way for theming available. By "refactor soon after it" I guess you mean the idea of having it generated from tokens package? The idea is good, but that won't be easy :)

Reply:
This `getTokens` function is documented as a function to generate themes, which is ultimately what we're doing here. We're "creating a new theme" that is basically the same theme but with the tokens associated with CSS Vars instead of direct values. That seems very hacky to me. By "refactor soon after this" I mean multiple things:
- Don't have a function named `getTokens` that receives one schema and returns the merge of that schema with other default ones. That is simply not clear.
- Don't need to recreate the theme and iterate over the keys multiple times (this is related to what I commented [here](https://github.com/kiwicom/orbit/pull/4087#discussion_r1361821347) about going around in circles with object iterations)

Ofc I don't have a proposed solution for now, and I agree it's not straightforward, but I believe we can simplify or at least improve readability (and maintenance) here.

Reply:
Yep, it can be solved that way only in the tokens package, here it's needed unfortunately to have it like that. I am trying to simplify this as much as possible for a minimal working solution.

Reply:
Yes, yes, I am not in favor of refactoring it now, let's not complicate it more. But please gather the ideas / feedback you have and the comments we have here so that we can consider a major refactor in the not-so-distant future